.pre-commit-config.yaml

```diff
@@ -43,50 +47,35 @@ repos:
   #         '--removed-in', '4.0',
   #         '--deprecated-in', '4.1',
   #       ]
-  # - repo: https://github.com/adamchainz/django-upgrade
-  #   rev: 1.7.0
-  #   hooks:
-  #     - id: django-upgrade
-  #       args: [--target-version, '4.0']
   - repo: https://github.com/adamchainz/django-upgrade
     rev: 1.12.0
     hooks:
-    - id: django-upgrade
-      args: [--target-version, "4.1"]
+      - id: django-upgrade
+        args: [--target-version, "4.1"]
+        alias: autoformat
   - repo: https://github.com/asottile/pyupgrade
     rev: v3.3.1
     hooks:
       - id: pyupgrade
-        args: [--py39-plus]
+        args: [--py38-plus]
+        alias: autoformat
   - repo: https://github.com/psf/black
     rev: 23.1.0
     hooks:
       - id: black
+        alias: autoformat
   - repo: https://github.com/asottile/blacken-docs
     rev: 1.13.0
     hooks:
       - id: blacken-docs
-        additional_dependencies:
-          - black==21.7b0
-  - repo: https://github.com/pycqa/isort
-    rev: 5.12.0
-    hooks:
-      - id: isort
-  - repo: https://github.com/PyCQA/flake8
-    rev: 6.0.0
-    hooks:
-      - id: flake8
-        additional_dependencies:
-          - flake8-bugbear
-          - flake8-comprehensions
-          - flake8-print
-          - flake8-no-pep420
-          - flake8-tidy-imports
-  - repo: https://github.com/codespell-project/codespell
-    rev: v2.2.2
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
```

Comment:
I think it's great that you use `ruff`, I've also started using it instead of `flake8` and `isort`, but it also replaces `pyupgrade` which you can remove.
https://github.com/charliermarsh/ruff#pyupgrade-up

Reply:
@pauloxnet thank you for the tip. I struggle with what ruff can lint vs. replace (like black{ish} support). Not sure if you have any public configs yet, but this is going to prompt me to check out some of your public repos again.